Refactoring HTML, CSS, View, Markup

magsout opened the pull request:
Normalise, cleaned, and refactor CSS/HTML/View …
 - removed useless component (SuitCSS)
 - normalise each CSS Component
 - normalise ClassName (e.g: IssueList -> ListIssue ...)
 - Split View in multiple file
 - Removed #id

We have a code base that grows a little every day. There is some legacy code that also happens. so I wanted with this PR, cleaned, added what I learned from the beginning that I work on this project. It's not finished yet, but I wanted to start by sending this PR.

I know, there is lot files changed. Hard to review (my apologies in advance). But when I started to edit the files, I started with a css file, and markup etc .. It was hard to make only small changes. But change is not going to break anything (I have fixed testing course)


What I have to do:

- [x] Issue (Component and Page)
- [x] LabelEditor 
- [x] Label
- [x] Fixed test 
- [x] Fixed warning variables
- [x] Replaced old custom media 
- [x] Thanks Page
- [x] CSSFixme
- [x] Error Page
- [x] removed dependencies between test (JS), lib : added js-* (as possible)
- [x] Form

Review needed. R? @miketaylr @karlcow @hallvors @tagawa

magsout commented:
I know, there is lot files changed. So hard to review (my apologies in advance). But when I started to edit the files, I started, then a css file, and markup etc .. It was hard to make only small changes. But change is not going to break anything (I have fixed testing course)

miketaylr commented on webcompat/static/css/development/components/UI.css:
```diff
@@ -0,0 +1,79 @@
+
+html {
+  -ms-text-size-adjust: 100%;
+  -webkit-text-size-adjust: 100%;
```
No `-moz-text-size-adjust`? :p (I guess the webkit alias will just work anyways, in the future...)

magsout replied:
Ah good catch.

miketaylr commented on webcompat/static/css/development/base/variable.css:
```diff
@@ -1,47 +0,0 @@
-/*----------Variable----------*/
```
Just to understand, variables are now defined in a `root` selector in each "module"?

magsout replied:
Yes I want to scope locally each component with this own variable.

miketaylr replied:
:+1:

miketaylr commented on webcompat/static/css/development/components/style-variable.css:
```diff
@@ -0,0 +1,57 @@
+/*----------Variable----------*/
+
+/* base */
+:root {
```
Oh, here they are. So these are "base" variables? And the ones in each module/file are "local" (or something)?

magsout replied:
Exactly

miketaylr commented on webcompat/static/css/development/components/style-variable.css:
```diff
@@ -0,0 +1,57 @@
+@custom-media --wc-max990px all and (max-width: 61.87em);
+@custom-media --wc-min990px all and (min-width: 61.88em);
+
+@custom-media --viewport-1180px all and (max-width: 73.75em);
```
We use a `--viewport-450px` in a few places, where is that defined?

magsout replied:
Yeah I didn't replace yet all custom-media. I'm not a big fan of the new name I choose. We need something simple, semantic and easy to remember.

magsout replied:
@miketaylr I use this https://github.com/cssrecipes/custom-media-queries#available-custom-media-queries for Grid, I can continue to use this convention, but we have lot of breakpoint, so If you have an idea. Or maybe, juste I remove all and use `@media all and (max/min-width: )`

miketaylr commented on webcompat/templates/issue.html:
```diff
@@ -1,135 +1,120 @@
+          <script type="text/template" id="comment-tmpl">
           <div class="wc-Comment-avatar">
-            <img src="<%= reporterAvatar %>" alt="<%= reporter %>">
+            <img src="<%= avatarUrl %>" alt="<%= commenter %>">
```
Note to self: this is kind of confusing, we have `reporterAvatar` and `avatarUrl` vs `reporter` and `commenter`. That could be more consistent.

magsout replied:
Hum why I overide this O_O. I didn't remember do it...

miketaylr replied:
You probably didn't --- I think the diff is just getting confused with line changes (I think you just moved some stuff around). The note was more for me. :)

miketaylr commented:
I didn't build or anything, but so far so good!

miketaylr commented:
(let me know if you want help updating any broken tests)

magsout commented:
@miketaylr thanks for your quick review. Just need to continu my todolist.

magsout commented on webcompat/templates/issue/issue-header.html:
```diff
@@ -0,0 +1,20 @@
+  <!-- Issue Date -->
+  <div class="wc-IssueDetail-create">
+    <script type="text/template" id="metadata-tmpl">
+    <div class="wc-Tag wc-Tag--label wc-Tag--<%= stateClass %>"><%= issueState %></div>
```
@miketaylr do you know how I can use macro  like this: `{{ tag('filter', <%= stateClass %>) }}` 

If I do this, I have an error.. 

poke @karlcow

miketaylr replied:
Hmm, I think we might be able to get around the templating error with the `{%raw%} blah {%endraw%}` in the macro definition: http://jinja.pocoo.org/docs/dev/templates/#escaping

But maybe it won't work because stateClass is defined in JS and you're hitting a reference error before the macro is even run.

`{{ tag('filter', <%= stateClass %>) }}` maybe possible to define in such a way to do `{{ tag('filter', 'stateClass') }}` (pass in stateClass as a string)?

magsout commented:
OK, travis is happy now :+1:

miketaylr commented:
@magsout  ready for review?

magsout commented:
@miketaylr not yet, I have to finished Form. Hum, maybe better to add linter in another PR ?

magsout commented:
@miketaylr  I think finish tonight or tomorrow. What do you think to put on staging for everybody can review the design? Check that I have not created regression?

miketaylr commented:
(oops, forgot about the checklist up top)

Yeah, maybe linter in another PR to keep it features more "atomic", if that makes sense.

miketaylr commented:
> What do you think to put on staging for everybody can review the design? Check that I have not created regression?

Sounds good! Just let me know when to deploy it. Also no rush -- I just wanted to make sure I wasn't sitting around ignoring reviews!

magsout commented:
@miketaylr Neeed u ;) I break something but I don"t know what. In fact all test passed, but since I handle homepage, issue List failed.. Any thought ?

miketaylr commented:
(just gonna tackle any test failures on a per-commit basis, if that's OK. </lazy)

miketaylr commented:
Hmm, I don't see any issue list failures when I run locally -- there was the one with the homepage form failure (which I just threw a `sleep` at). Ready for review?

magsout commented:
@miketaylr yes 👍

magsout commented:
r? @miketaylr @karlcow @hallvors 

I have not touched the design. The only thing really visible is the size of container. It's bigger than before. But if that does not suit you, I can reduce it. All elements are in EM and REM for accessibility. I fixed zoom bug on the forms. I''ve added classes JS- * on the markup as possible to avoid breaking the test when there is change in the DOM. 

Again, sorry for this review with a lot of change

## Before
![capture d ecran 2016-01-26 a 09 29 35](https://cloud.githubusercontent.com/assets/1997108/12575903/804dafa8-c410-11e5-96d0-9b8a1f6c8188.png)

## After
![capture d ecran 2016-01-26 a 09 29 41](https://cloud.githubusercontent.com/assets/1997108/12575904/804efa3e-c410-11e5-8885-b60c520ee6c0.png)

miketaylr commented:
The wider display looks god to me.

miketaylr commented on tests/functional/comments-auth.js:
```diff
@@ -126,7 +126,7 @@ define([
         .end()
         .sleep(500)
-        .findAllByCssSelector('.js-issue-comment:not(.wc-Comment--form)')
+        .findAllByCssSelector('.js-issue-comment')
```
Thank you for updating all of these -- it's tedious work, but now they won't break. :sunglasses:

miketaylr commented on tests/functional/index-non-auth.js:
```diff
@@ -30,7 +30,7 @@ define([
       return this.remote
         .get(require.toUrl(url('/')))
-        .findByCssSelector('.wc-Navbar-link').getVisibleText()
+        .findByCssSelector('.js-Navbar-link').getVisibleText()
```
Do we want to have a consistent style for `js-` classes/ids? Some of them are all lowercase, some mixed-case. Maybe it doesn't really matter...

magsout replied:
Hum yes, my convention is namespace-Component-children--modifier. I think it's good to have a constistent ClassName convention. I don't know what do you want for JS ? js-Navbar-link is equal to the className for the markup wc-Navbar-link

miketaylr replied:
> js-Navbar-link is equal to the className for the markup wc-Navbar-link

That sounds like a good convention then. ^_^

magsout replied:
I have made  some mistake

miketaylr commented on tests/functional/user-activity-auth.js:
```diff
@@ -47,23 +47,23 @@ define([
+        .findByCssSelector('.js-issue-list:nth-child(1) > div:nth-child(1) > p:nth-child(2)').getVisibleText()
         .then(function(text) {
           assert.match(text, /comments:\s\d+$/i, 'Issue should display number of comments');
           assert.match(text, /^Opened:\s\d{4}\-\d{2}\-\d{2}.+/, 'Issue should display creation date');
```
Tests all look good. :100:

magsout commented on tests/functional/index-non-auth.js:
```diff
@@ -20,7 +20,7 @@ define([
       return this.remote
         .get(require.toUrl(url('/')))
-        .findByCssSelector('.wc-Hero-title').getVisibleText()
+        .findByCssSelector('.js-hero-title').getVisibleText()
```
Error js-Hero-title is better ...

miketaylr commented on webcompat/static/css/development/components/comment.css:
```diff
@@ -1,25 +1,37 @@
-/*------Comment------*/
+/** @define Issue */
```
`@define Issue`, is `Comment` here renamed, or is this a mistake?

miketaylr commented on webcompat/static/css/development/components/issue-list.css:
```diff
@@ -1,65 +1,127 @@
+    background-color: transparent;
+  }
+
+  /* Statut */
```
Typo?

miketaylr commented on webcompat/static/css/development/components/issue-list.css:
```diff
@@ -1,65 +1,127 @@
-/*------Issue List------*/
+/** @define IssueItem */
```
`IssueList`?

miketaylr commented on webcompat/templates/list-issue/issue-list.html:
```diff
@@ -1,34 +1,34 @@
-<div class="wc-content js-issue-list">
+<div class="wc-UIContent js-list-issue">
```
Are we renaming all instances of `IssueList` or `issue-list` to `ListIssue` and `list-issue`? Or is there a difference between the two now?

miketaylr commented on webcompat/templates/list-issue.html:
```diff
@@ -1,24 +1,27 @@
+      <div class="wc-ListIssue-sorting">
+        {% include "list-issue/list-issue-sorting.html" %}
+      </div>
+      {% include "list-issue/issue-list.html" %}
```
So here `list-issue.html` is including `list-issue/issue-list.html`. What's the list-issue component for? I think I'm just confused. ^_^

miketaylr commented:
r+ from me with the small issues you've pointed out. I'm still not sure what the difference between issue-list and list-issue is, but I'm sure you'll tell me. :)

magsout commented:
@miketaylr I changed `ListIssue` in `SearchIssue`  is more relevant. I changed all classes `js-*` according to convention seen in the issue. I have not touched the `js-` class which was not related to a component

miketaylr commented:
@magsout ah, ok sounds good. I say feel free to merge when you're happy. HUGE improvements. :heart:

magsout commented:
Any thought ? @karlcow @hallvors @calexity @tagawa ?

karlcow commented:
yes more white space around the issues gives it more air. Breathable. 
@magsout did you test on mobile the new version?

💌　good to send.

tagawa commented:
Agree, I like the wider layout as well. I thought having the labels showing (android, opera, etc.) was helpful but maybe that's a personal preference.

magsout commented:
@tagawa what do you mean about label ? I don't understand

tagawa commented:
@magsout In your "before" screenshot the browser labels are visible, but in the "after" screenshot they've been removed.

magsout commented:
@tagawa ah yes, don't worries, labels are still here. The Second screenshot is take on my localhost and the issues have not labels.

tagawa commented:
Ah, my misunderstanding. All good then, thanks.

magsout commented:
@karlcow yes 

![screen recording 2016-01-29 at 09 48 pm](https://cloud.githubusercontent.com/assets/1997108/12687760/22209c32-c6d2-11e5-8577-2b7b4a1b130a.gif)
